main.tf

```diff
@@ -64,81 +64,72 @@
-  provider = "aws.peer"
+data "aws_region" "remote" {
+  provider = "aws.remote"
 }
 
-data "aws_route_tables" "this_vpc_rts" {
-  provider = "aws.this"
-  vpc_id   = "${var.this_vpc_id}"
+/*
+data "aws_route_table" "local_vpc_rt" {
+  provider = "aws.local"
+  vpc_id   = "${var.local_vpc_id}"
 }
 
-data "aws_route_tables" "peer_vpc_rts" {
-  provider = "aws.peer"
-  vpc_id   = "${var.peer_vpc_id}"
+data "aws_route_table" "remote_vpc_rt" {
+  provider = "aws.remote"
+  vpc_id   = "${var.remote_vpc_id}"
+  subnet_id = "subnet-05e007e946ff14426"
+}
+*/
+
+data "aws_vpc" "local" {
+  provider = "aws.local"
+  id       = "${var.local_vpc_id}"
+}
+
+data "aws_vpc" "remote" {
+  provider = "aws.remote"
+  id       = "${var.remote_vpc_id}"
 }
 
 # Create a route
-resource "aws_route" "this_rt" {
-  provider                  = "aws.this"
-  route_table_id            = "${var.this_main_route_table_id}"
-  destination_cidr_block    = "${var.peer_cidr_block}"
+resource "aws_route" "local_rt" {
+  provider                  = "aws.local"
+  route_table_id            = "${coalesce(var.local_main_route_table_id, data.aws_vpc.local.main_route_table_id)}"
+  destination_cidr_block    = "${var.remote_subnet_range}"
   vpc_peering_connection_id = "${aws_vpc_peering_connection.peering.id}"
 }
 
 ## Create a route
-resource "aws_route" "peer_rt" {
-  provider                  = "aws.peer"
-  route_table_id            = "${var.peer_main_route_table_id}"
-  destination_cidr_block    = "${var.this_cidr_block}"
+resource "aws_route" "remote_rt" {
+  provider                  = "aws.remote"
+  route_table_id            = "${coalesce(var.remote_main_route_table_id, data.aws_vpc.remote.main_route_table_id)}"
+  destination_cidr_block    = "${var.local_subnet_range}"
   vpc_peering_connection_id = "${aws_vpc_peering_connection.peering.id}"
 }
 
 resource "aws_vpc_peering_connection" "peering" {
-  provider    = "aws.this"
-  vpc_id      = "${var.this_vpc_id}"
-  peer_vpc_id = "${var.peer_vpc_id}"
-  peer_region = "${data.aws_region.peer.name}"
+  provider    = "aws.local"
+  vpc_id      = "${var.local_vpc_id}"
+  peer_vpc_id = "${var.remote_vpc_id}"
+  peer_region = "${data.aws_region.remote.name}"
 
   tags = "${merge(var.tags, map("Name", "VPC Peering between default and bursting"))}"
 }
 
 # Accepter's side of the connection.
-resource "aws_vpc_peering_connection_accepter" "peer" {
-  provider                  = "aws.peer"
+resource "aws_vpc_peering_connection_accepter" "remote" {
+  provider                  = "aws.remote"
   vpc_peering_connection_id = "${aws_vpc_peering_connection.peering.id}"
   auto_accept               = true
 
   tags = "${merge(var.tags, map("Name", "Accepter"))}"
 }
-
-resource "aws_security_group_rule" "this_sg" {
-  provider    = "aws.this"
-  type        = "ingress"
-  from_port   = 0
-  to_port     = 65535
-  protocol    = "all"
-  cidr_blocks = ["${var.peer_cidr_block}"]
-
-  security_group_id = "${var.this_security_group_id}"
-}
-
```

Comment:
Hey @fatz @geekbass, Will this removal still give us the single terraform apply ability?

Reply:
Yes!